client/state/initial-state.js

```diff
@@ -25,6 +25,15 @@ const HOUR_IN_MS = 3600000;
 export const SERIALIZE_THROTTLE = 5000;
 export const MAX_AGE = 7 * DAY_IN_HOURS * HOUR_IN_MS;
 
+/**
+ * In-memory copy of persisted state.
+ *
+ * We load from browser storage into this cache on boot, and initialize state
+ * from it, rather than asynchronously reading from browser storage for every
+ * persisted reducer.
+ */
+let stateCache = {};
```

Comment:
Do we only need this during boot? Would a WeakMap make more sense here?

Reply:
Yes, we only need this during boot, but "boot" can be kind of an odd concept when loading reducers individually. Once we modularise state, loading a particular reducer might happen at an arbitrary point in time as the user navigates through the application (e.g. only loading a stats-related reducer once the user navigates to stats).

I'm not sure that using a `WeakMap` here would translate into meaningful memory savings, but we can certainly use one defensively, as it shouldn't introduce any issues by itself 👍

Reply:
Actually, now that I think about it further, I think a `WeakMap` would not work at all, precisely because we do need to preserve the state that was read in memory until that arbitrary point in time. If we really want to clear memory, we could do so by cleaning it after it's read from the cache, but I'm not sure if that's worth the risk.

Reply:
Sounds good 👍